web/src/lib/components/photos-page/asset-grid.svelte

```diff
@@ -81,18 +82,34 @@
     element.scrollBy(0, event.detail.heightDelta);
   }
 
-  const navigateToPreviousAsset = async () => {
-    const prevAsset = await assetStore.getPreviousAssetId($viewingAsset.id);
-    if (prevAsset) {
-      assetViewingStore.setAssetId(prevAsset);
+  const handlePrevious = async () => {
+    const previousAsset = await assetStore.getPreviousAssetId($viewingAsset.id);
+    if (previousAsset) {
+      assetViewingStore.setAssetId(previousAsset);
     }
+
+    return !!previousAsset;
```

Comment:
Can you help explain the logic here?

Reply:
Just returns a boolean value to indicate if one was found or not. Makes it possible to fallback to navigating to the previous asset and again falling back to closing the viewer if there is not another asset to show.

Basically, when you archive an asset and it should advance, what if there is no next asset? Now we can know that and show the Prius one instead. Double !! just concerts it to a boolean.